modules/porous_flow/src/userobjects/PorousFlowBrineCO2.C

```diff
@@ -37,9 +39,42 @@ PorousFlowBrineCO2::PorousFlowBrineCO2(const InputParameters & parameters)
 {
   // Check that the correct FluidProperties UserObjects have been provided
   if (_co2_fp.fluidName() != "co2")
-    mooseError("Only a valid CO2 FluidProperties UserObject can be provided in co2_fp");
+    mooseError(name(), ": a valid CO2 FluidProperties UserObject must be provided in co2_fp");
   if (_brine_fp.fluidName() != "brine")
-    mooseError("Only a valid brine FluidProperties UserObject can be provided in brine_fp");
+    mooseError(name(), ": a valid brine FluidProperties UserObject must be provided in brine_fp");
+
+  // Set the number of phases and components, and their indexes
+  _num_phases = 2;
+  _num_components = 3;
+  _gas_phase_number = 1 - _aqueous_phase_number;
+  _gas_fluid_component = 3 - _aqueous_fluid_component - _salt_component;
+
+  // Check that _aqueous_phase_number is <= total number of phases
+  if (_aqueous_phase_number >= _num_phases)
+    mooseError(
+        name(),
+        ": the value provided in liquid_phase_number is larger than the possible number of phases ",
+        _num_phases);
+
+  // Check that _aqueous_fluid_component is <= total number of fluid components
+  if (_aqueous_fluid_component >= _num_components)
+    mooseError(name(),
+               ": the value provided in liquid_fluid_component is larger than the possible  number "
+               "of fluid components",
+               _num_components);
+
+  // Check that the salt component index is not identical to the liquid fluid component
+  if (_salt_component == _aqueous_fluid_component)
+    mooseError(
+        name(),
+        ": the values provided in salt_component and liquid_fluid_component must be different");
+
+  // Check that _salt_component is <= total number of fluid components
+  if (_salt_component >= _num_components)
+    mooseError(name(),
```

Comment:
Tip: We recently added "paramError()" and "paramWarning()" macros in MooseObject.C. You can use those as drop-ins where you want to report information about input file problems to the user (you have to supply the name of the affected parameter). MOOSE will print the line number of the problematic parameter to help the user find the issue faster. You and Andy may want to start putting these in.

Reply:
Thanks for the tip. We have a lot of these input parameter checks, so anything to make it easier to find problems is appreciated!